core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -229,6 +244,58 @@ class SharedAllocationRecord<void, void> {
       const SharedAllocationRecord* const root, const bool detail);
 };
 
+template <class MemorySpace>
+class SharedAllocationRecordCommon : public SharedAllocationRecord<void, void> {
+ private:
+  using derived_t     = SharedAllocationRecord<MemorySpace, void>;
+  using record_base_t = SharedAllocationRecord<void, void>;
+  derived_t& self() { return *static_cast<derived_t*>(this); }
```

Comment:
```suggestion
  derived_t& self() { return static_cast<derived_t&>(this); }
```

Reply:
That doesn't compile. I think you mean `return static_cast<derived_t&>(*this);`, but I seem to remember there are reasons to prefer the former in CRTP. I could be wrong, though.

Reply:
Yes I mean `*this`.  I'd like to hear the reason but I am not blocking.

Reply:
I haven't yet been able to think of a reason why these would ever do different things.